File: cognipy/cognipylib/cognipy/ICogniPySvr.java
package cognipy;

import cognipy.models.*;
import java.util.*;

public interface ICogniPySvr
{
	void LoadCnl(String filename, boolean loadAnnotations, boolean materialize, boolean modalCheck);
	java.lang.Iterable<AnnotationResult> GetAnnotationsForSignature(java.lang.Iterable<String> cnlEntities);
	HashMap<String, ConstraintResult> GetConstraints(ArrayList<String> descriptions);
	ArrayList<CogniPyStatement> ToCNLStatementList();
	HashMap<String, InstanceDescription> DescribeInstances(String query);
	ArrayList<String> GetSubConceptsOf(String cnlName, boolean direct);
	ArrayList<String> GetInstancesOf(String cnlName, boolean direct);
	void KnowledgeInsert(String text, boolean loadAnnotations, boolean materialize);
	void KnowledgeDelete(String text, boolean materialize);

	void LoadCnlFromString(String cnl, boolean loadAnnotations, boolean materialize, boolean modalCheck);
	void LoadRdf(String uri, boolean loadAnnotations, boolean materialize, boolean modalCheck);
	void LoadRdfFromString(String rdf, boolean loadAnnotations, boolean materialize, boolean modalCheck);
	String ToCNL(boolean includeAnnotations);
	ArrayList<String> ToCNLList(boolean includeAnnotations);
	ArrayList<String> GetSuperConceptsOf(String cnlName, boolean direct);
}

File: docsrc/jupyter_notebooks/BugReports/cognipy/CogniPyLib/ICogniPySvr.cs
using CogniPy.models;
using System.Collections.Generic;

namespace CogniPy
{
    public interface IModularizer
    {
        void LoadCnlFromString(string cnl);
        void LoadFromReasoner(ICogniPySvr client);
        void KnowledgeInsert(string text);
        void KnowledgeDelete(string text);
        IEnumerable<string> GetModule(IEnumerable<string> signature);
    }

    public interface ICogniPySvr
    {
        void LoadCnl(string filename, bool loadAnnotations, bool materialize, bool modalCheck, bool throwOnError);
        IEnumerable<AnnotationResult> GetAnnotationsForSignature(IEnumerable<string> cnlEntities);
        Dictionary<string, ConstraintResult> GetConstraints(List<string> descriptions);
        List<CogniPyStatement> ToCNLStatementList();
        Dictionary<string, InstanceDescription> DescribeInstances(string query);
        List<string> GetSubConceptsOf(string cnlName, bool direct);
        List<string> GetInstancesOf(string cnlName, bool direct);
        void KnowledgeInsert(string text, bool loadAnnotations, bool materialize);
        void KnowledgeDelete(string text, bool materialize);

        void LoadCnlFromString(string cnl, bool loadAnnotations, bool materialize, bool modalCheck, bool throwOnError);
        void LoadRdf(string uri, bool loadAnnotations, bool materialize, bool modalCheck, bool throwOnError);
        void LoadRdfFromString(string rdf, bool loadAnnotations, bool materialize, bool modalCheck, bool throwOnError);
        string ToCNL(bool includeAnnotations);
        List<string> ToCNLList(bool includeAnnotations);
        List<string> GetSuperConceptsOf(string cnlName, bool direct);
    }

}
